nixos/modules/services/x11/display-managers/slim.nix

```diff
@@ -126,7 +122,12 @@ in
 
   ###### implementation
 
-  config = mkIf cfg.enable {
+  config = mkIf (elem dmcfg.enable [ "slim" "auto" ]) {
```

Comment:
It sounds like we do want to have a `services.xserver.displayManager.slim.enabled` (with an extra `d`) internal option, in order to make these abstraction modular.

In this case we want the `auto` display manager to be an alias for enabling the `slim` display manager as a back-end implementation while being added in the same unique list as `slim`.  Having `enabled` option checks would allow us to work-around the uniqueness of the `enum` type.